ocaml/utils/compilation_unit.ml

```diff
@@ -326,6 +562,7 @@ let which_cmx_file desired_comp_unit ~accessed_by : t =
        current pack. *)
     desired_comp_unit
   else
+    let () = assert (not (is_instance desired_comp_unit)) in
```

Comment:
It would require reformatting the whole block. (There should really be a `begin` and `end` around the whole `else` but Mark decided a while ago that we can't have nice things.)